### PR TITLE
updated with test

### PR DESCRIPTION
**Pull request by dami-moj**

### CODE REVIEW

File: dev.env [hunk 1]
```diff
@@ -2,10 +2,11 @@
...
 LANDING_FOLDER='s3://dami-test-bucket786567/de-intro/land/'
 RAW_HIST_FOLDER='s3://dami-test-bucket786567/de-intro/raw_history/'
 CURATED_FOLDER='s3://dami-test-bucket786567/de-intro/curated/'
-METADATA_FOLDER='s3://dami-test-bucket786567/de-intro/metadata/'
-MOJAP_EXTRACTION_TS=1704451200
+METADATA_FOLDER='s3://dami-test-bucket786567/de-intro/metadata/' 
+MOJAP_EXTRACTION_TS=1704451202
 MOJAP_IMAGE_VERSION='v0.0.1'
-TABLES='peoples'
+TABLES='peoples'
```
Comment (Thomas-Hirsch) [inline dev.env:12]: for what it's worth, "people" is the plural of person; "peoples" is the plural of _people_, in the sense of "group of people of the same ethnic group". I'm not sure exactly what your data contains, [but it's worth understanding the difference](https://www.britannica.com/dictionary/eb/qa/What-is-the-difference-between-people-and-peoples-).

**Review by Thomas-Hirsch — APPROVED** [review]
The tests pass and it seems reasonable to my eye. I've added a few comments that aren't directly related to this PR but are worth considering.

For the future, it would be useful to add a description to pull requests to summarise what the changes mean. This would help both the people reviewing it understand what to focus on, as well as you when understanding the reasonings behind the change when looking back in the future.

File: scripts/run.py [hunk 2]
```diff
@@ -1,45 +1,64 @@
...
-        print("Check the process")
\ No newline at end of file
+        df = cast_columns_to_correct_types(df, metadata)
+        df = add_mojap_columns_to_dataframe(
+            df,
+            settings.MOJAP_IMAGE_VERSION,
+            settings.MOJAP_EXTRACTION_TS,
+        )
+
+        db_dict = {
```
Comment (Andy-Cook) [inline scripts/run.py:48]: I can understand the use of a dictionary object here, however for readability of functions called as part of this functionality, passing these values as arguments to the functions would be a more preferable approach.

File: scripts/run.py [hunk 3]
```diff
@@ -1,45 +1,64 @@
...
+    update_metadata,
     load_data_from_s3,
     cast_columns_to_correct_types,
     add_mojap_columns_to_dataframe,
     write_curated_table_to_s3,
     move_completed_files_to_raw_hist,
 )
 
-def run_data_pipeline():
+config = {
```
Comment (Andy-Cook) [inline scripts/run.py:15]: Defining this dict would probably be better placed after the main-lock (`if __name__ == "__main__":`) and then be passed to the `main()` function explicitly.

File: scripts/functions.py [hunk 4]
```diff
@@ -154,27 +155,23 @@ def update_metadata(metadata: Metadata) -> Metadata:
...
+) -> pd.DataFrame:
     """Casts a single column in a DataFrame to a specific type."""
     if column_name not in df.columns:
-        if column_type == "timestamp(ms)":
-            df[column_name] = pd.NaT
-        elif column_type == "string":
-            df[column_name] = ""
-        else:
-            df[column_name] = pd.NA
+        df[column_name] = (
```
Comment (Andy-Cook) [inline scripts/functions.py:165]: My preference is towards your previous approach to doing this, even if the "one-liner' approach might be a little more 'pythonic'. 
Where you've got if, elif, else statements I find it easier to read in a 'switch block' format. But very much personal preference here.

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -20,7 +20,7 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install -r requirements.lock
```
[the remark this answers is not in the document]
Comment (dami-moj) [reply .github/workflows/test.yml:23]: It was originally modified

File: dev.env (hunk 1 above)
Comment (dami-moj) [reply dev.env:12]: This was just a name for the table, considering that I am trying to replicate the approach used in [airflow-laa-extractor](https://github.com/moj-analytical-services/airflow-laa-extractor)

**Pull request merged**